### PR TITLE
Add contract name map

### PR DESCRIPTION
**Pull request by danfinlay**
Moved off `ethereum-contract-icons` to `eth-contract-metadata` to enable richer metadata than just icons before we get too invested in that new repo.

Looks like this:

screen shot 2017-05-26 at 10 23 48 am

### CODE REVIEW

**Review by kumavis — COMMENTED** [review]
I recommend not using the "in" operator bc it does not check for truthiness

`a in {a: undefined}` evaluates to true

**Review by kumavis — COMMENTED** [review]
There's a comment that says nickname keys must be lowercased that is no longer true I think

**Comment by flyswatter** [comment]
done.

File: ui/lib/contract-namer.js [hunk 1]
```diff
@@ -6,13 +6,18 @@
  */
 
 // Nickname keys must be stored in lower case.
-const nicknames = {}
+const contractMap = require('eth-contract-metadata')
+const ethUtil = require('ethereumjs-util')
 
 module.exports = function (addr, identities = {}) {
+  const checksummed = ethUtil.toChecksumAddress(addr)
+  if (contractMap.checksummed && contractMap[checksummed].name) {
```
Comment (kumavis) [inline ui/lib/contract-namer.js]: Checksummed is a variable not a property

**Review by kumavis — CHANGES_REQUESTED** [review]

File: ui/lib/contract-namer.js [hunk 2]
```diff
@@ -6,13 +6,18 @@
  */
 
 // Nickname keys must be stored in lower case.
```
Comment (kumavis) [inline ui/lib/contract-namer.js]: Remove or fix comment

File: ui/lib/icon-factory.js [hunk 3]
```diff
@@ -43,11 +43,12 @@ IconFactory.prototype.generateNewIdenticon = function (address, diameter) {
 // util
 
 function iconExistsFor (address) {
-  return (address in iconMap) && isValidAddress(address)
+  return (address in contractMap) && isValidAddress(address) && ('logo' in contractMap[address])
```
Comment (kumavis) [inline ui/lib/icon-factory.js]: Still seeing the in operator

Comment (flyswatter) [inline ui/lib/icon-factory.js]: oh I changed it in the other place.  This is a 2nd place.

**Review by kumavis — APPROVED** [review]

**Pull request merged**